external-crates/move/crates/move-package/src/compilation/build_plan.rs

```diff
@@ -130,12 +136,20 @@ impl BuildPlan {
     }
 
     /// Compilation process does not exit even if warnings/failures are encountered
-    pub fn compile_no_exit<W: Write>(&self, writer: &mut W) -> Result<CompiledPackage> {
+    pub fn compile_no_exit<W: Write>(
+        &self,
+        writer: &mut W,
+        modify_compiler: impl FnOnce(Compiler) -> Compiler,
+    ) -> Result<CompiledPackage> {
         self.compile_with_driver(writer, |compiler| {
-            let (files, units_res) = compiler.build()?;
+            let (files, units_res) = modify_compiler(compiler).build()?;
```

Comment:
Nit: this name is a little odd, though I do not have a better suggestion off the top of my head.